src/eth_provider/provider.rs

```diff
@@ -524,6 +525,38 @@ where
         let gas_price = kakarot_contract.get_base_fee().call().await?.base_fee;
         Ok(into_via_wrapper!(gas_price))
     }
+
+    async fn block_receipts(&self, block_id: Option<BlockId>) -> EthProviderResult<Option<Vec<TransactionReceipt>>> {
+        let block_id = block_id.unwrap_or(BlockId::Number(BlockNumberOrTag::Latest));
+        match block_id {
+            BlockId::Number(maybe_number) => {
+                let block_number = match maybe_number {
+                    BlockNumberOrTag::Latest => self.block_number().await?,
+                    BlockNumberOrTag::Earliest => U64::ZERO,
+                    BlockNumberOrTag::Number(number) => U64::from(number),
+                    BlockNumberOrTag::Pending => self.block_number().await? + U64::from(1_u64),
+                    BlockNumberOrTag::Finalized => {
+                        return Err(EthProviderError::MethodNotSupported(
+                            "Block tag variant Error: Cannot query finalized blocks in Kakarot network. Change BlockTag to 'latest'".to_string()
+                        ));
+                    }
+                    BlockNumberOrTag::Safe => {
+                        return Err(EthProviderError::MethodNotSupported(
+                            "Block tag variant Error: Cannot query safe blocks in Kakarot network. Change BlockTag to 'latest'".to_string()
+                        ));
+                    }
+                };
```

Comment:
there is a helper for this at the end of the file, you could update the helper `tag_into_block_number` and then use it here.